### PR TITLE
Updating endorspy module

### PR DESCRIPTION
**Pull request by aidaanva**
## PR checklist

Updating the module for endorspy to be linked to the new version that calculates not only more versions of the percent on target but also clonality from samtools flagstats

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/endorspy/meta.yml [hunk 1]
```diff
@@ -20,14 +20,15 @@ input:
...
       type: file
-      description: output of samtools flagstat in a text file
-  - stats_optional:
+      description: output of samtools flagstat in a txt file, assumes no quality filtering nor duplicate removal performed
+  - stats_qualityfiltered:
       type: file
-      description: output of a second samtools flagstat run in a text file. If two files are supplied,
-        the mapped reads of the second file is divided by the total reads in the first, since it
-        assumes that the <samplefile.stats> are related to the same sample. Useful after BAM filtering.
+      description: output of samtools flagstat in a txt file, assumes some form of quality or length filtering has been performed, must be provided with at least one of the options -r or -d
```
Comment (jfy133) [inline modules/nf-core/endorspy/meta.yml:28]: Should we add a check in `script` that if `stats_qualityfiltered && !optionalraw && !deduplicated` it fails?

File: tests/modules/nf-core/endorspy/test.yml [hunk 2]
```diff
@@ -1,7 +1,85 @@
...
+  - path: output/endorspy/test_percent_on_target_mqc.json
+    md5sum: 1940461b14e6150138ca71ef7d395440
+  - path: output/samtools/test.bam
+    md5sum: 1bfa483a2991037107732ccedef8ae07
+  - path: output/samtools/test.bam.bai
+    md5sum: 077fa5c7206bf5cd43a08788c79ab9f2
+  - path: output/samtools/test.filtered.flagstat
+    md5sum: eed2b068e1eb7a3d748f094738e26ad0
+  - path: output/samtools/test.flagstat
+    md5sum: 4f7ffd1e6a5e85524d443209ac97d783
```
Comment (jfy133) [inline tests/modules/nf-core/endorspy/test.yml:25]: All of these are not necessary, you just need to check for the JSOn (with md5sum) and the `versions.yml` (no md5sum check, just the path)

**Review by jfy133 — CHANGES_REQUESTED** [review]
One suggestion and one cleanup :)

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: modules/nf-core/endorspy/meta.yml (hunk 1 above)
Comment (aidaanva) [reply modules/nf-core/endorspy/meta.yml:28]: it does fail

Comment (jfy133) [reply modules/nf-core/endorspy/meta.yml:28]: Sorry I mean we should add a warning if that conditio nis detected

Comment (aidaanva) [reply modules/nf-core/endorspy/meta.yml:28]: You mean in the description?

File: tests/modules/nf-core/endorspy/test.yml (hunk 2 above)
Comment (aidaanva) [reply tests/modules/nf-core/endorspy/test.yml]: would I just remove this?

**Comment by aidaanva** [comment]
Please recheck @jfy133 @scarlhoff @jbv2

**Review by jfy133 — APPROVED** [review]

**Comment by jfy133** [comment]
@nf-core-bot fix linting

**Pull request merged**